art/estimators/classification/pytorch.py

```diff
@@ -848,10 +925,29 @@ def get_activations(
         else:  # pragma: no cover
             raise TypeError("Layer must be of type str or int")
 
+        def get_feature(name):
+            # the hook signature
+            def hook(model, input, output):  # pylint: disable=W0622,W0613
+                self._features[name] = output
+
+            return hook
+
+        if not hasattr(self, "_features"):
+            self._features: Dict[str, torch.Tensor] = {}
+            # register forward hooks on the layers of choice
+        if layer not in self._features.keys():
+            interim_layer = dict([*self._model._model.named_modules()])[  # pylint: disable=W0212,W0622,W0613
+                self._layer_names[layer_index]
+            ]
+            interim_layer.register_forward_hook(get_feature(self._layer_names[layer_index]))
+
         if framework:
             if isinstance(x, torch.Tensor):
-                return self._model(x)[layer_index]
-            return self._model(torch.from_numpy(x).to(self._device))[layer_index]
+                self._model(x)
```

Comment:
Shouldn't this also use x_preprocessed instead of x?

Reply:
I tried to maintain the same consistency as base code, which handles framework=True, this way. Please check - https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/dev_1.9.0/art/estimators/classification/pytorch.py

Reply:
Sure thing. Seems like it was addressed by PR 1471